src/syft/lib/python/primitive_factory.py

```diff
@@ -1,14 +1,15 @@
-from typing import Union
+from typing import Union, Optional
 
 from ...decorators import syft_decorator
 from .primitive_interface import PyPrimitive
+from ...core.common import UID
 
 
 class PrimitiveFactory:
     @staticmethod
     @syft_decorator(typechecking=True)
     def generate_primitive(
-        value: Union[int, float, bool, complex, None]
+        value: Union[int, float, bool, complex, None], id: Optional[UID] = None
     ) -> PyPrimitive:
         from .int import Int
```

Comment:
Q: Can't we move those outside?

Reply:
It would be great to move these imports before we merge.

Reply:
It is a circular dependency, let it be for now and fix it when we have it all.

Reply:
Mmm - in my experience circular dependencies only get worse the longer you wait. We are under a deadline so I'm tempted this time - but technical_debt+=1